### PR TITLE
Add path of failing pubspec.yaml on YamlException

### PR DESCRIPTION
**Pull request by matanlurey**
This makes debugging https://github.com/dart-lang/bazel/issues/61 easier.

### CODE REVIEW

**Comment by natebosch** [comment]
Can you give an example of what the error looks like with the latest bazel? I don't think we should see the stack trace anymore - what do we see instead?

File: lib/src/bazelify/build.dart [hunk 1]
```diff
@@ -101,7 +102,15 @@ class BuildFile {
   /// - Every package generates _exactly one_ dart_library
   /// - Some packages generate 1 or more dart_vm_binary or dart_web_application
   static Future<BuildFile> fromPackageDir(String path) async {
-    final pubspec = await Pubspec.fromPackageDir(path);
+    Pubspec pubspec;
+    try {
+      pubspec = await Pubspec.fromPackageDir(path);
+    } on YamlException catch (e) {
+      throw new YamlException(
```
Comment (natebosch) [inline lib/src/bazelify/build.dart]: how about handling this in Pubspec.fromPackageDir?

**Review by natebosch — CHANGES_REQUESTED** [review]

**Comment by matanlurey** [comment]
@natebosch The error is "nicer" but you still lost context on what path was being traversed.

I'll make the change to move the try/catch into `fromPackageDir`.

**Comment by matanlurey** [comment]
Closing for now. I'll re-open if this is still an issue.

**Pull request closed without merging**